SU-HSDO/suhumsci
# STN-674: Callout Box Component

# READY FOR REVIEW

## Summary
Create a Callout Box component, the ability to float a text area content within an immersive story.

West D7 Example: https://west.stanford.edu/news/blogs/and-the-west-blog/2020/fight-plague-local-governments-lean-trust-could-they-lose-it

HumSci D8 example: https://news.stanford.edu/2021/01/27/diseases-history-intertwined/

JIRA Card: https://sparkbox.atlassian.net/browse/STN-674

## Need Review By (Date)
asap

## Urgency
medium

## Steps to Test
1. npm run test
2. Create a Flexible Page test page.
3. In the components paragraphs area you should see Callout Box in the list of components to choose from, add one.
4. The Callout Box only has two fields, the text area and the alignment selection, by default it will be selected to the Right.
5. Create a callout box with the default right alignment and one with the default left alignment.
6. On small/mobile sized screens the Callout Box should be 100% width and stack like the other components above it.
7. On medium to desktop sized screens the Callout Box should float within the text area below it and be a 40% width as requested in the card it will not be 100% on desktop size even if it is floating above a non-text area component.
8. Place the callout box above and below any other components and it should not float or interact with the other component types, unless it is a text area below it.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
